CHANGELOG.md

```diff
@@ -9,6 +9,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ### Added
 - Implement `FromStr` and `ToString` traits for public sodium types (#318)
+- Sparse list index - a new data structure (#312)
```

Comment:
A thing to consider: do we want a more detailed changelog? Maybe "...a new data structure similar to `ListIndex`, but it may contain indexes without elements".